# origin: Debian, #162190
# cset: n/a
# inclusion: not submitted
# description: remove period check for executables
# revision date: 2004-09-04

diff -urN kernel-source-2.4.26/fs/isofs/inode.c kernel-source-2.4.26-1/fs/isofs/inode.c
--- kernel-source-2.4.26/fs/isofs/inode.c	2002-11-29 10:53:15.000000000 +1100
+++ kernel-source-2.4.26-1/fs/isofs/inode.c	2002-11-30 15:49:27.000000000 +1100
@@ -1190,14 +1193,6 @@
 		inode->i_mode = inode->i_sb->u.isofs_sb.s_mode;
 		inode->i_nlink = 1;
 	        inode->i_mode |= S_IFREG;
-		/* If there are no periods in the name,
-		 * then set the execute permission bit
-		 */
-		for(i=0; i< de->name_len[0]; i++)
-			if(de->name[i]=='.' || de->name[i]==';')
-				break;
-		if(i == de->name_len[0] || de->name[i] == ';')
-			inode->i_mode |= S_IXUGO; /* execute permission */
 	}
 	inode->i_uid = inode->i_sb->u.isofs_sb.s_uid;
 	inode->i_gid = inode->i_sb->u.isofs_sb.s_gid;
